### ISSUE TITLE
not reading aws environmental variables

### ISSUE THREAD

**Issue by ekimia** [blender/Rome#102]
So it fails, given `~/.aws/credentials` doesn't exist. My build:
https://travis-ci.org/zerofinancial/relay/builds/269878346

    **Rome version:** v0.12.0.31
    **OS and version:** [Enter OS name and version here]

**Comment by blender** [comment]
Thanks, I'll look into it

**Comment by ekimia** [comment]
thanks!

**Comment by blender** [comment]
@ekimia this works for me if I do 

```
export AWS_ACCESS_KEY_ID=<some access key>
export AWS_SECRET_ACCESS_KEY=<some secret>
```

**Comment by ekimia** [comment]
update on this. Travis is setup properly and the variables do exist in my `build.sh` script. As a workaround I generated the `.aws/config` file programmatically

**Comment by blender** [comment]
@ekimia but do they exist in `compile.sh` ?

**Comment by ekimia** [comment]
yup!

**Comment by blender** [comment]
@ekimia ok Thanks. The problem then seems to be much deeps than Rome. 

I'll try to make a minimal experiment and see. Thanks for your understanding and for finding a workaround in the mean time.

**Comment by ekimia** [comment]
no problem, let me know if you need help

**Comment by blender** [comment]
@itsthejb 

Unless you're trying to use the "engine" feature, Rome doesn't launch any child process. It's using system provided access to env variables via http://hackage.haskell.org/package/base-4.12.0.0/docs/System-Environment.html#v:lookupEnv 

I doubt this is bug in a core Haskell library. Rather something is interfering with your VM's env variables. In Travis for example reading env works just fine.

**Comment by itsthejb** [comment]
Debugging on a BuddyBuild remote session:

```
export AWS_REGION=eu-central-1
export AWS_ACCESS_KEY_ID=keyblahblah
export AWS_SECRET_ACCESS_KEY=secretblahblah
```

Then, `rome download --skip-local-cache` results in `The AWS Access Key Id you provided does not exist in our records.` for all dependencies

**Comment by blender** [comment]
try `AWS_REGION= ... AWS_ACCESS_KEY_ID=... AWS_SECRET_ACCESS_KEY=... rome ...`

**Comment by blender** [comment]
Also, if you are using Minio instead of S3 you need AWS_ENDPOINT

**Comment by itsthejb** [comment]
We aren't using Minio, that was my error since I found that issue in the issue list first.

Interesting discoveries, using version `v0.20.0.56` on BuddyBuild:

```
AWS_REGION=eu-central-1 \
AWS_ACCESS_KEY_ID=foo \
AWS_SECRET_ACCESS_KEY=bar \
rome list
```

List dependencies... whereas:

```
AWS_REGION=eu-central-1 \
AWS_ACCESS_KEY_ID=foo \
AWS_SECRET_ACCESS_KEY=bar \
rome download
```

Complains about the key. Currently the bucket is (even) completely public (for testing purposes!), so this is strange

**Comment by blender** [comment]
Try running this on CI: https://mega.nz/#!ik1xVArD!7mq-Dtzb07yMZccNSc6rwjYB_enRw6t8XVxDNWEqcWY

<detail> Source Code

```haskell
module Main where

import System.Environment

main :: IO ()
main = do
  aws_region <- lookupEnv "AWS_REGION"
  aws_id <- lookupEnv "AWS_ACCESS_KEY_ID"
  aws_secret <- lookupEnv "AWS_SECRET_ACCESS_KEY"
  print aws_region
  print aws_id
  print aws_secret
```

</detail>

**Comment by itsthejb** [comment]
Can check that in a moment. However, more relevantly, perhaps; we use a wrapper around `aws` CLI here, which writes `~/.aws/credentials` with the following keys:

```
[default]
aws_access_key_id = ...
aws_secret_access_key = ...
aws_session_token = ...
aws_security_token = ...
```

I suspect for this to work I potentially need to pass all of them...? Also, please forgive my relative AWS ignorance, not my speciality 😢

**Comment by itsthejb** [comment]
^ Got it now, I think! I just need to pass _all_ of those as env vars:

- `AWS_REGION`
- `AWS_ACCESS_KEY_ID`
- `AWS_SECRET_ACCESS_KEY`
- `AWS_SESSION_TOKEN`
- `AWS_SECURITY_TOKEN`

Generally caused by my ignorance. Perhaps a docs improvement would be to list all of those keys as potential candidates, and something along the lines of; "if these are listed in your profile in credentials, you need to pass them all to Rome"?

**Comment by blender** [comment]
I have no idea what AWS_SESSION_TOKEN or AWS_SECURITY_TOKEN are doing. I doubt I'm reading them on the latest version.

**Comment by itsthejb** [comment]
Now confirmed that I have something _basically_ working, although not enough to exactly confirm what the issue is. I'll update again when I know. Other minor issue I'm now seeing is Rome produces no output when running via the build server. I also checked redirecting stderr to stdout to see if BuddyBuild only outputs stdout (seems unlikely). Could there be something sensitive that might prevent Rome producing output? Currently makes it quite hard to work out what's going on

**Comment by blender** [comment]
No, there is no redirection of std input or output. You can use `-v` for force more verbose info
